### PR TITLE
fetchFollowing and fetchFollowers created as methods, lookupChannel now takes in a optional type, which can be a parent_url or id

### PR DESCRIPTION
**Pull request by MrKevinOConnell**
New methods:
fetchUserFollowingV2 -> https://docs.neynar.com/reference/following-v2
fetchUserFollowersV2 -> https://docs.neynar.com/reference/followers-v2

Updates
lookupChannel -> https://docs.neynar.com/reference/channel-details
now takes in a optional type, which can be a parent_url or id

misc:
- channel_follow_context description fixed for ValidateFrameRequest
- parent description improved for PostCastReqBody
- ChannelType is now a type

- fetchUserFollowingV2 with fid set to 3
Screenshot 2024-04-29 at 11 24 57 AM
- fetchUserFollowersV2 with fid set to 3
Screenshot 2024-04-29 at 11 25 35 AM

### CODE REVIEW

File: src/neynar-api/neynar-api-client.ts [hunk 1]
```diff
@@ -2249,6 +2250,66 @@ export class NeynarAPIClient {
...
+  * @param {FollowSortType} [options.sortType] Sort type for retrieving followers. Default is FollowSortType.DescChron
+  * @param {number} [options.limit] Number of results to retrieve (default 20, max 100)
+  * @param {string} [options.cursor] Pagination cursor.
+  *  omit this parameter for the initial request.
+  * 
+  * @returns {Promise<UsersResponse>} A promise that resolves to a `UsersResponse` object,
+  * 
+  * @example
+  * // Example: Retrieve followers for a user
+  * import { FollowSortType } from "@neynar/nodejs-sdk";
```
Comment (Shreyaschorge) [inline src/neynar-api/neynar-api-client.ts:2268]: `FollowSortType` need to be exported from `src/neynar-api/index.ts` in order to import it like this

File: src/neynar-api/neynar-api-client.ts [hunk 2]
```diff
@@ -2249,6 +2250,66 @@ export class NeynarAPIClient {
...
+  * // Example: Retrieve follows for a user
+  * import { FollowSortType } from "@neynar/nodejs-sdk";
+  * 
+  * client.fetchUserFollowingV2(3,{limit: 5,viewerFid: 3}).then(response => {
+  * console.log('User Follows:', response);
+  * });
+  * 
+  * For more information, refer to the [Neynar documentation](https://docs.neynar.com/reference/following-v2).
+*/
+  public async fetchUserFollowingV2(
```
Comment (Shreyaschorge) [inline src/neynar-api/neynar-api-client.ts:2306]: @manan19 Should we name it like this given that v1 is called `fetchUserFollowing`, if both methods returning the same data converted to v2 objects, then I think the name make sense

cc @MrKevinOConnell to confirm v1 and v2 responses

**Review by Shreyaschorge — CHANGES_REQUESTED** [review]
@MrKevinOConnell Minor changes the rest looks good.

File: src/neynar-api/neynar-api-client.ts (hunk 2 above)
Comment (MrKevinOConnell) [reply src/neynar-api/neynar-api-client.ts:2306]: i believe fetchUserFollowing is returning v1 objects whereas fetchUserFollowingV2 is returning v2 objects @Shreyaschorge

File: src/neynar-api/neynar-api-client.ts (hunk 1 above)
Comment (MrKevinOConnell) [reply src/neynar-api/neynar-api-client.ts:2268]: done!

File: src/neynar-api/neynar-api-client.ts (hunk 2 above)
Comment (Shreyaschorge) [reply src/neynar-api/neynar-api-client.ts:2306]: @MrKevinOConnell That's true, what I meant was apart from v1 and v2 objects is there anything else that has changed?

Comment (MrKevinOConnell) [reply src/neynar-api/neynar-api-client.ts:2315]: no they are the same @Shreyaschorge 👍🏻

**Comment by Shreyaschorge** [comment]
@MrKevinOConnell Please resolve merge conflicts

**Review by Shreyaschorge — CHANGES_REQUESTED** [review]

**Comment by manan19** [comment]
Combine with #109 and bump the version to 1.22.0

**Review by Shreyaschorge — APPROVED** [review]
LGTM

**Pull request merged**